src/session/session_dhandle.c

```diff
@@ -575,9 +575,22 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
              */
 
             if (first_snapshot_time != snapshot_time || ds_time > snapshot_time ||
-              hs_time > snapshot_time || stable_time > snapshot_time || oldest_time > snapshot_time)
-                ret = __wt_set_return(session, EBUSY);
-            else {
+              hs_time > snapshot_time || stable_time > snapshot_time ||
+              oldest_time > snapshot_time) {
+                /*
+                 * When a system wide checkpoint is not running, we can hang here. This situation
+                 * can occur when a checkpoint on a single file has been performed by a bulk
+                 * operation and no system wide checkpoint has been done since then. It is also
+                 * possible to end up here when opening the history store using a checkpoint cursor
+                 * internally, we want to wait for the inconsistency to get resolved in this case.
```

Comment:
This code is not waiting, it is either returning EBUSY or NOTFOUND. This comment should not talk about what other code may or may not do. Maybe something like:
"When opening the history store using a checkpoint cursor internally, we need to return EBUSY so that ..."

Reply:
In fact, if `ret` is set to `EBUSY`, it stays in the `while` loop, that's why it is "waiting", see [here](https://github.com/wiredtiger/wiredtiger/pull/8924/files/17110caadb4400186a3050813ed5e1434614a486#diff-86364bae325f76f9e7ae1af1acc5a8008692138c0387c17bf52abe055989c177L641).